### PR TITLE
Builderv2 content

### PR DESCRIPTION
**Pull request by fepitre**

### CODE REVIEW

**Comment by fepitre** [comment]
PipelineRetry

File: template_rpm/09_cleanup.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# shellcheck source=qubesbuilder/plugins/template_rpm/distribution.sh
+source "${TEMPLATE_CONTENT_DIR}/distribution.sh"
+
+rm -f "${INSTALL_DIR}"/var/lib/rpm/__db.00* "${INSTALL_DIR}"/var/lib/rpm/.rpm.lock
```
Comment (marmarek) [inline template_rpm/09_cleanup.sh:27]: This lacks getting dbpath dynamically (rpm --eval ...).

File: template_rpm/3rd_party_software/RPM-GPG-KEY-rpmfusion-free-fedora-36 [hunk 2]
```diff
@@ -0,0 +1,29 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```
Comment (marmarek) [inline template_rpm/3rd_party_software/RPM-GPG-KEY-rpmfusion-free-fedora-36:1]: Those were symlinks to the actual keys (the same for multiple releases). Any specific reason for committing separate copies?

File: template_rpm/add_3rd_party_software.sh [hunk 3]
```diff
@@ -0,0 +1,112 @@
+#!/bin/bash -e
+# vim: set ts=4 sw=4 sts=4 et :
+
+# shellcheck source=qubesbuilder/plugins/template_rpm/distribution.sh
```
Comment (marmarek) [inline template_rpm/add_3rd_party_software.sh:4]: same here

File: template_rpm/distribution.sh [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+    if [ "${DIST#fc}" != "${DIST}" ]; then
+        DIST_NAME="fedora"
+        DIST_VER="${DIST#fc}"
+    elif [ "${DIST#centos-stream}" != "${DIST}" ]; then
+        DIST_NAME="centos-stream"
+        DIST_VER="${DIST#centos-stream}"
+    fi
+fi
+
+# shellcheck source=qubesbuilder/plugins/template/scripts/functions.sh
```
Comment (marmarek) [inline template_rpm/distribution.sh:22]: same here

File: template_rpm/distribution.sh [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+source "${TEMPLATE_SCRIPTS_DIR}/functions.sh" >/dev/null
+# shellcheck source=qubesbuilder/plugins/template/scripts/umount-kill
```
Comment (marmarek) [inline template_rpm/distribution.sh:24]: same here

**Review by marmarek — CHANGES_REQUESTED** [review]

File: template_rpm/04_install_qubes.sh [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        kver="$(basename "$kver")"
+        chroot_cmd dracut -f -a "qubes-vm" \
+            "/boot/initramfs-${kver}.img" "${kver}" || RETCODE=1
+    done
+    chroot_cmd grub2-install --target=i386-pc "$dev" || RETCODE=1
+    chroot_cmd grub2-mkconfig -o /boot/grub2/grub.cfg || RETCODE=1
+    if [ "0${IS_LEGACY_BUILDER}" -eq 1 ]; then
+        fuser -kMm "${INSTALL_DIR}"
+    else
+        fuser -kMm @BUILDER_DIR@/mnt
```
[the remark this answers is not in the document]
Comment (fepitre) [reply template_rpm/04_install_qubes.sh:125]: I think it should be ok but I need to check that in a separate PR if you are ok. Notably, after having split the content in qubes-builderv2.

File: template_rpm/3rd_party_software/RPM-GPG-KEY-rpmfusion-free-fedora-36 (hunk 2 above)
Comment (fepitre) [reply template_rpm/3rd_party_software/RPM-GPG-KEY-rpmfusion-free-fedora-36:1]: No, it's certainly a failure during backporting or similar.

**Review by marmarek — APPROVED** [review]

**Pull request merged**